### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/Sourcetrail/commit/d867bf6270364608ee6da7a2a664d65153f98637#commitcomment-48074522

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/Sourcetrail/commit/89e393f55baa54edff93a82547574ec2f5e86bdf

Note: this PR does not include the action. If you're interested in running a spell check on every PR and push, that can be offered separately.

### CODE REVIEW

File: src/lib/utility/commandline/commands/CommandlineCommandConfig.cpp [hunk 1]
```diff
@@ -90,7 +90,7 @@ void CommandlineCommandConfig::setup()
 		"logging-enabled,l", po::value<bool>(), "Enable file/console logging <true/false>")(
 		"verbose-indexer-logging-enabled,L",
 		po::value<bool>(),
-		"Enable additional log of abstract syntax tree during the indexing. <true/false> WARNINIG "
+		"Enable additional log of abstract syntax tree during the indexing. <true/false> WARNING "
```
Comment (jsoref) [inline src/lib/utility/commandline/commands/CommandlineCommandConfig.cpp:93]: This probably changes the CLI

File: docs/documentation/index.html [hunk 2]
```diff
@@ -2068,7 +2068,7 @@ <h1>Project Setup</h1>
 
 					<h2>Choose a Source Group Type</h2>
 
-					<p>When creating a new Source Group you can choose between different methods of setup depending on your project language. Please have a look at our <a href="#Choosingtherightprojectsetup">QUCICK START GUIDE section</a> for help on choosing the right Source Group type.</p>
+					<p>When creating a new Source Group you can choose between different methods of setup depending on your project language. Please have a look at our <a href="#ChoosingTheRightProjectSetup">QUICK START GUIDE section</a> for help on choosing the right Source Group type.</p>
```
Comment (jsoref) [inline docs/documentation/index.html:2071]: This anchor doesn't exist, nor did it exist when this content was added...

Comment (mlangkabel) [reply docs/documentation/index.html:2071]: Thanks for noticing. We will correct this in a different commit.

**Review by mlangkabel — CHANGES_REQUESTED** [review]

File: script/deploy_windows.sh [hunk 3]
```diff
@@ -89,13 +89,13 @@ echo -e "$INFO Building the project"
 
 if [ $RUN_CODE_SIGNING = true ]; then
 	echo -e "$INFO Signing the app"
-	signtool.exe sign //sha1 $CODE_SIGNING_THUMBPRINT //fd sha256 //t "http://timestamp.verisign.com/scripts/timstamp.dll" //d "Sourcetrail ${DOTTED_VERSION_STRING}" //du "https://www.sourcetrail.com/" //v build/Release/app/Sourcetrail.exe
+	signtool.exe sign //sha1 $CODE_SIGNING_THUMBPRINT //fd sha256 //t "http://sha256timestamp.ws.symantec.com/sha256/timestamp" //d "Sourcetrail ${DOTTED_VERSION_STRING}" //du "https://www.sourcetrail.com/" //v build/Release/app/Sourcetrail.exe
```
[the remark this answers is not in the document]
Comment (jsoref) [reply script/deploy_windows.sh:92]: `timstamp` is an 8.3 thing, but Verisign's `timstamp` endpoint died ages ago. You can pick any replacement you like. Symantec is roughly of similar caliber to Verisign, hence the choice. But it doesn't really matter to me.

It's also fine to split this out, as it's a behavior change -- albeit from "using a service that doesn't exist at all":
```
curl -vvv http://timestamp.verisign.com/scripts/timstamp.dll
*   Trying 216.168.248.38...
* TCP_NODELAY set
* Connection failed
* connect to 216.168.248.38 port 80 failed: Network is unreachable
* Failed to connect to timestamp.verisign.com port 80: Network is unreachable
* Closing connection 0
curl: (7) Failed to connect to timestamp.verisign.com port 80: Network is unreachable
```
to one that could actually work...

File: src/lib_gui/CMakeLists.txt [hunk 4]
```diff
@@ -37,8 +37,8 @@ add_files(
 	qt/element/code/QtCodeFileTitleBar.h
 	qt/element/code/QtCodeFileTitleButton.cpp
 	qt/element/code/QtCodeFileTitleButton.h
-	qt/element/code/QtCodeNavigateable.cpp
-	qt/element/code/QtCodeNavigateable.h
+	qt/element/code/QtCodeNavigable.cpp
```
[the remark this answers is not in the document]
Comment (jsoref) [reply src/lib_gui/CMakeLists.txt:40]: I have a supplemental tool which handles renames and takes the same input as the general tool.

It mostly happens for Java classes.

For a while I dealt w/ it manually, but enough java projects led to me automating it.

File: src/external/tinyxml/tinyxml.h [hunk 5]
```diff
@@ -576,7 +576,7 @@ class TiXmlNode : public TiXmlBase
 	#endif
 
 	/** Add a new node related to this. Adds a child past the LastChild.
-		Returns a pointer to the new object or NULL if an error occured.
+		Returns a pointer to the new object or NULL if an error occurred.
```
[the remark this answers is not in the document]
Comment (jsoref) [reply src/external/tinyxml/tinyxml.h:579]: Tool glitch, I had already set things to exclude `src/external` -- dropped.

**Review by mlangkabel — CHANGES_REQUESTED** [review]

File: bin/test/data/JavaIndexSampleProjectsTestSuite/JavaSymbolSolver060/src/java-symbol-solver-core/com/github/javaparser/symbolsolver/javaparsermodel/contexts/ContextHelper.java [hunk 6]
```diff
@@ -45,19 +45,19 @@ private ContextHelper() {
 
     public static Optional<MethodUsage> solveMethodAsUsage(TypeDeclaration typeDeclaration, String name,
                                                            List<Type> argumentsTypes, TypeSolver typeSolver,
-                                                           Context invokationContext, List<Type> typeParameters) {
+                                                           Context invocationContext, List<Type> typeParameters) {
```
[the remark this answers is not in the document]
Comment (jsoref) [reply bin/test/data/JavaIndexSampleProjectsTestSuite/JavaSymbolSolver060/src/java-symbol-solver-core/com/github/javaparser/symbolsolver/javaparsermodel/contexts/ContextHelper.java:48]: Yeah, my tool screamed at me when I excluded the path. It took me a while to catch the discrepancy.

File: script/deploy_windows.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (jsoref) [reply script/deploy_windows.sh:92]: Moved to #1164

**Review by mlangkabel — APPROVED** [review]

**Comment by mlangkabel** [comment]
Thanks @jsoref for putting in all the effort!

**Pull request merged**